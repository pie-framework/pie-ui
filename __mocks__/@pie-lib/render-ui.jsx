import React from 'react';

export const Feedback = () => <div>Feedback</div>;

export const indicators = {
  Correct: () => <div>Correct</div>,
};

export const PreviewPrompt = () => <div>PreviewPrompt</div>;

export const color = {
  text: jest.fn(),
  correct: jest.fn(),
  incorrect: jest.fn(),
  disabled: jest.fn(),
  primary: jest.fn(),
  primaryLight: jest.fn(),
  primaryDark: jest.fn(),
  primaryText: jest.fn(),
  secondary: jest.fn(),
  secondaryLight: jest.fn(),
  secondaryDark: jest.fn(),
  secondaryText: jest.fn(),
  background: jest.fn(),
};
