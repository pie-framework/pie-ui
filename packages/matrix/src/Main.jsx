import React from 'react';
import PropTypes from 'prop-types';
import {PreviewLayout} from '@pie-lib/render-ui';
import Matrix from './Matrix';

class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onSessionChange: PropTypes.func
  };

  static defaultProps = {
    model: {},
    session: {},
  };

  render() {
    const {model, onSessionChange, session} = this.props;
    return (
      <PreviewLayout>
        <Matrix
          {...model}
          session={session}
          onSessionChange={onSessionChange}
        />
      </PreviewLayout>
    );
  }
}

export default Main;
