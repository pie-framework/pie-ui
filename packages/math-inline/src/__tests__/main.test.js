import * as React from 'react';
import Main from '../main';
import { shallowChild } from '@pie-lib/test-utils';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { Feedback } from '@pie-lib/render-ui';

xdescribe('Main', () => {
  const defaultProps = {
    model: {
      config: {
        responses: [],
        id: '1',
        element: 'math-inline',
        feedback: {
          correct: {
            type: 'none',
            default: 'Correct'
          },
          partial: {
            type: 'none',
            default: 'Nearly'
          },
          incorrect: {
            type: 'none',
            default: 'Incorrect'
          }
        }
      }
    },
    onSessionChange: jest.fn(),
    session: {}
  };

  let wrapper;
  let component;

  beforeEach(() => {
    wrapper = shallowChild(Main, defaultProps, 1);
  });

  it('renders correctly', () => {
    component = wrapper();

    expect(component.find(CorrectAnswerToggle).length).toEqual(1);
    expect(component.find(Feedback).length).toEqual(0);
    //
    // expect(component.state()).toEqual({
    //   session: {
    //     answers: {
    //       1: [false, false],
    //       2: [false, false],
    //       3: [false, false],
    //       4: [false, false]
    //     }
    //   },
    //   showCorrect: false,
    //   shuffledRows: [{
    //     id: 1,
    //     title: 'Question Text 1',
    //     values: [false, false]
    //   }, {
    //     id: 2,
    //     title: 'Question Text 2',
    //     values: [false, false]
    //   }, {
    //     id: 3,
    //     title: 'Question Text 3',
    //     values: [false, false]
    //   }, {
    //     id: 4,
    //     title: 'Question Text 4',
    //     values: [false, false]
    //   }]
    // })
  });
});
