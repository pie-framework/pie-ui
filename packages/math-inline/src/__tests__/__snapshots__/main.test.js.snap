// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Math-Inline Main render renders correctly with snapshot 1`] = `
<Main
  classes={
    Object {
      "blockContainer": "Main-blockContainer-17",
      "blockMath": "Main-blockMath-20",
      "blockResponse": "Main-blockResponse-18",
      "collapsible": "Main-collapsible-9",
      "content": "Main-content-8",
      "correct": "Main-correct-15",
      "correctAnswerShown": "Main-correctAnswerShown-14",
      "expression": "Main-expression-11",
      "incorrect": "Main-incorrect-16",
      "inputAndKeypadContainer": "Main-inputAndKeypadContainer-12",
      "keypadTooltip": "Main-keypadTooltip-4",
      "keypadTooltipPopper": "Main-keypadTooltipPopper-5",
      "main": "Main-main-6",
      "mainContainer": "Main-mainContainer-1",
      "responseContainer": "Main-responseContainer-10",
      "showCorrectness": "Main-showCorrectness-13",
      "title": "Main-title-7",
      "toggle": "Main-toggle-19",
      "tooltip": "Main-tooltip-2",
      "tooltipPopper": "Main-tooltipPopper-3",
    }
  }
  model={
    Object {
      "config": Object {
        "customKeys": Array [
          "\\\\left(\\\\right)",
          "\\\\frac{}{}",
          "x\\\\frac{}{}",
        ],
        "element": "math-inline",
        "equationEditor": "everything",
        "expression": "\\\\text{A family sized box contains} {{response}} \\\\text{less than} {{response}} \\\\text{times the number  }  \\\\frac{3}{6}=\\\\frac{ {{response}} }{4} + \\\\frac{ {{response}} }{4}",
        "feedback": Object {
          "correct": Object {
            "default": "Correct",
            "type": "none",
          },
          "incorrect": Object {
            "default": "Incorrect",
            "type": "none",
          },
          "partial": Object {
            "default": "Nearly",
            "type": "none",
          },
        },
        "question": "<p>The binomial <math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"> <mi>x</mi><mo>&#8722;</mo><mn>2</mn> </math> is a factor of the polynomial below.</p><p><span class=\\"equation-block\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"> <mi>f</mi><mo stretchy=\\"false\\">(</mo><mi>x</mi><mo stretchy=\\"false\\">)</mo><mo>=</mo><msup> <mi>x</mi> <mn>3</mn> </msup> <mo>+</mo><msup> <mi>x</mi> <mn>2</mn> </msup> <mo>+</mo><mi>n</mi><mi>x</mi><mo>+</mo><mn>10</mn> </math> </span></p><p>What is the value of <span class=\\"variable\\">n</span>? Use the on-screen keyboard&#160;to type the correct answer in the box.</p>",
        "responseType": "Advanced Multi",
        "responses": Array [
          Object {
            "alternates": Object {},
            "answer": "\\\\frac{3}{6}=\\\\frac{1}{2}",
            "validation": "literal",
          },
        ],
      },
      "correctness": Object {},
      "element": "math-inline",
      "id": "1",
    }
  }
  onSessionChange={[MockFunction]}
  session={Object {}}
/>
`;
